frappe/database/query.py

```diff
@@ -111,11 +111,14 @@ def apply_filters(
 			self.apply_dict_filters(filters)
 
 		elif isinstance(filters, (list, tuple)):
-			for filter in filters:
-				if isinstance(filter, (str, int, Criterion, dict)):
-					self.apply_filters(filter)
-				elif isinstance(filter, (list, tuple)):
-					self.apply_list_filters(filter)
+			if all(isinstance(d, (str, int)) for d in filters):
+				self.apply_dict_filters({"name": ("in", filters)})
```

Comment:
Is this sugar really required? Also it's not supported in other places to best of my knowledge so again inconsistencies. 

```
{"name": ["ABC", "XYZ"]}
``

Add tests for any sugar you add!

Reply:
Also get_all does weird argument gymnastic so list filters will become fields to select 

https://github.com/frappe/frappe/blob/53f81549078e3bdc3fcb5d29109eabc4457fd42c/frappe/model/db_query.py#L122-L124

Reply:
> Is this sugar really required?

I don't like it either. But the previous implementation of `frappe.qb.engine.get_query` supported it, so I added it for feature parity. Should I remove it?

Reply:
> Also get_all does weird argument gymnastic so list filters will become fields to select

Let's not do this mistake again? `get_all` and `get_query` are different APIs. `get_all` can continue to support it's "weird magic" while using `get_query` under the hood.